sklearn/datasets/kddcup99.py

```diff
@@ -0,0 +1,414 @@
+        data = np.random.permutation(data)
+
+    if subset == 'SF' or subset == 'http' or subset == 'smtp':
+        # select all samples with positive logged_in attribute:
+        s = data[:, 11] == 1
+        data = np.c_[data[s, :11], data[s, 12:]]
+
+        a0 = np.array([data[:, 0]]).T
+        a2 = np.array([data[:, 2]]).T
+        a4 = np.array([data[:, 4]]).T
+        a5 = np.array([data[:, 5]]).T
+        a6 = np.array([data[:, -1]]).T
+            
+        a0 = np.log(np.asarray(a0 + 0.1, dtype=float))
+        a4 = np.log(np.asarray(a4 + 0.1, dtype=float))
+        a5 = np.log(np.asarray(a5 + 0.1, dtype=float))
+
+        if subset == 'http':
+            s = data[:, 2] == 'http'
+            data = data[s]  # np.c_[data[s, :2], data[s, 3:]]
+            data = np.concatenate((a0, a4, a5, a6), axis=1)
+
+
+        if subset == 'smtp':
+            s = data[:, 2] == 'smtp'
+            data = data[s]  # np.c_[data[s, :2], data[s, 3:]]
+            data = np.concatenate((a0, a4, a5, a6), axis=1)
+
+        if subset == 'SF':
+            data = np.concatenate((a0, a2, a4, a5, a6), axis=1)
+
+    target = data[:, -1]
+    data = data[:, :-1]
+
+    if shuffle:
+        rng = check_random_state(random_state)
+        indices = np.arange(target.shape[0])
+        rng.shuffle(indices)
+        data = data[indices]
+        target = target[indices]
+
+    return Bunch(data=data, target=target)
+
+
+def fetch_kddcup99(subset=None, data_home=None, download_if_missing=True,
+                   random_state=None, shuffle=False):
+
+    """Load the kddcup99 dataset, downloading it if necessary.
+
+    Parameters
+    ----------
+
+    data_home : string, optional
+        Specify another download and cache folder for the datasets. By default
+        all scikit learn data is stored in '~/scikit_learn_data' subfolders.
+
+    download_if_missing : boolean, default=True
+        If False, raise a IOError if the data is not locally available
+        instead of trying to download the data from the source site.
+
+    random_state : int, RandomState instance or None, optional (default=None)
+        Random state for shuffling the dataset.
+        If int, random_state is the seed used by the random number generator;
+        If RandomState instance, random_state is the random number generator;
+        If None, the random number generator is the RandomState instance used
+        by `np.random`.
+
+    shuffle : bool, default=False
+        Whether to shuffle dataset.
+
+    Returns
+    -------
+    dataset : dict-like object with the following attributes:
+
+    dataset.data : numpy array of shape (581012, 41) XXX
+        Each row corresponds to the 54XXX features in the dataset.
+
+    dataset.target : numpy array of shape (581012,) XXX
+        Each value corresponds to one of the 21 attack types or to the label
+        'normal.'.
+
+    dataset.DESCR : string
+        Description of the kddcup99 dataset.
```

Comment:
no :)

the sections names in docstring is fixed and you cannot add one. What you describe
in Returns Attributes should be in the one docstring of the dataset parameter.